File: reference/reference.net/db4o-reference-chapters/src/Db4objects.Db4odoc/Messaging/MessagingExample.cs
using System;
using Db4objects.Db4o;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Messaging;

namespace Db4objects.Db4odoc.Messaging
{

	public class MessagingExample 
	{
		private const string Db4oFileName = "reference.db4o";

        public static void Main(string[] Args)
        {
            ConfigureServer();
        }
        //end Main

		public static void ConfigureServer() 
		{
            IConfiguration configuration = Db4oFactory.NewConfiguration();
            configuration.ClientServer().SetMessageRecipient(new SimpleMessageRecipient());
			IObjectServer objectServer = Db4oFactory.OpenServer(configuration, Db4oFileName, 0);
            try 
			{
                IConfiguration clientConfiguration = Db4oFactory.NewConfiguration();
                // Here is what we would do on the client to send the message
                IMessageSender sender = clientConfiguration.ClientServer().GetMessageSender();
				IObjectContainer clientObjectContainer = objectServer.OpenClient(clientConfiguration);
				
				sender.Send(new MyClientServerMessage("Hello from client."));
				clientObjectContainer.Close();
			} 
			finally 
			{
				objectServer.Close();
			}
		}
		// end ConfigureServer 
	}

	public class SimpleMessageRecipient: IMessageRecipient
	{
		public void ProcessMessage(IMessageContext context,Object message) 
		{
			// message objects will arrive in this code block
			System.Console.WriteLine(message);
		}
	}
}

File: reference/reference.vb.net/Db4objects.Db4odoc/Messaging/MessagingExample.vb
' Copyright (C) 2004 - 2007 db4objects Inc. http://www.db4o.com 
Imports System
Imports Db4objects.Db4o
Imports Db4objects.Db4o.Config
Imports Db4objects.Db4o.Messaging


Namespace Db4objects.Db4odoc.Messaging

    Public Class MessagingExample
        Private Const Db4oFileName As String = "reference.db4o"

        Public Shared Sub ConfigureServer()
            Dim configuration As IConfiguration = Db4oFactory.NewConfiguration()
            configuration.ClientServer.SetMessageRecipient(New SimpleMessageRecipient())
            Dim objectServer As IObjectServer = Db4oFactory.OpenServer(configuration, Db4oFileName, 0)
            Try
                ' Here is what we would do on the client to send the message
                Dim clientConfiguration As IConfiguration = Db4oFactory.NewConfiguration()
                Dim sender As IMessageSender = clientConfiguration.ClientServer().GetMessageSender()
                Dim clientObjectContainer As IObjectContainer = objectServer.OpenClient(clientConfiguration)
                sender.Send(New MyClientServerMessage("Hello from client."))
                clientObjectContainer.Close()
            Finally
                objectServer.Close()
            End Try
        End Sub
        ' end ConfigureServer
    End Class

    Public Class SimpleMessageRecipient
        Implements IMessageRecipient
        Public Sub ProcessMessage(ByVal context As IMessageContext, ByVal message As Object) Implements IMessageRecipient.ProcessMessage
            ' message objects will arrive in this code block
            System.Console.WriteLine(message)
        End Sub
    End Class

End Namespace
